_posts/2020-03-18-how-to-use-agent-information-on-puppetserver.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: post
+title: "How to use agent-side information on the puppetserver"
+author: dev_el_ops
+categories:
+  - module
+  - development
+tags:
+  - modules
+  - puppetserver
+  - custom facts
+  - custom resources
+  - deferred
+---
+
+This question was posted on the [Puppet Community Slack](https://slack.puppet.com/) yesterday:
+
+> Is there any way like the below in puppet code we can use:
+> ```
+> $var=‘run comman locally on agent, if $?=0 echo “present”, else echo “absent”’
+> if $var == present {
+>   # my code
+> } else {
+>   # something else
+> }
+> ```
+(by user [IlovPuppet](https://app.slack.com/team/URGFPBS0N), edited for presentation)
+
+From my answer there, I wrote up a more detailed exposition here.
```

Comment:
I suggest that at this point in the blog, the "Why" of this is not yet answered. I would suggest adding a sentence or two explaining the kinds of things it enables, or why someone would want to use agent-side information.

To see what I mean,  How is IlovPuppet's problem not solved quickly and directly by a custom/external fact? Why do we need something other than a fact?

Reply:
I'll add a few words here. Part of the problem is that I don't know much about the motivations of people :-|
all I know about the technical reasons I've tried capturing below